providers/implementations/keymgmt/ecx_kmgmt.c

```diff
@@ -612,11 +733,31 @@ static void *ecx_gen(struct ecx_gen_ctx *gctx)
         if (!ossl_ed25519_public_from_private(gctx->libctx, key->pubkey, privkey,
                                               gctx->propq))
             goto err;
+#ifdef FIPS_MODULE
+        if (!fips_eddsa_pct(gctx->libctx,
+                            ECX_KEY_TYPE_ED25519,
+                            key->pubkey,
+                            privkey,
+                            key->keylen)) {
+            /* rsa pct aborts on fail, do the same. */
+            abort();
```

Comment:
should we abort? or should we return an error?

Reply:
abort() is what the RSA PCT does in this case. I just made eddsa do the same.